include/libtorrent/aux_/session_impl.hpp

```diff
@@ -169,19 +169,35 @@ namespace aux {
 		// had to retry binding with a higher port
 		int original_port = 0;
 
-		// this is typically set to the same as the local
-		// listen port. In case a NAT port forward was
-		// successfully opened, this will be set to the
-		// port that is open on the external (NAT) interface
-		// on the NAT box itself. This is the port that has
-		// to be published to peers, since this is the port
-		// the client is reachable through.
-		int tcp_external_port = 0;
-		int udp_external_port = 0;
+		// tcp_external_port and udp_external_port return the port which
+		// should be published to peers/trackers for this socket
+		// If there are active NAT mappings the return value will be
+		// the external port returned by the NAT router, otherwise the
+		// local listen port is returned
+		int tcp_external_port()
```

Comment:
It looks like every call to these two functions immediately casts the results to `uint16_t`. Do you think it would make sense to just return that type right away?

Reply:
There are a couple of uses without immediate casts in `session_impl::get_listen_port` and `session_impl::listen_port`. This hits on the larger issue of port numbers using a mix of `std::uint16_t` and `int`. My preference would be to make a separate PR to convert all port numbers to use `std::uint16_t`.